src/rgw/rgw_op.h

```diff
@@ -1065,7 +1065,7 @@ class RGWGetACLs : public RGWOp {
 
 class RGWPutACLs : public RGWOp {
 protected:
-  size_t len;
+  int len;
```

Comment:
Ugh, an unfortunate situation.  Seems you have a clear idea of what's going on. 
Would it be acceptable to check it situation and throw on failure, rather than perpetuate that, or is it better to continue in this way? (You'll know better than I do, but I certainly would rather see size_t when size_t is meant.)